agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -1226,6 +1226,42 @@ bool monitor_thread::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
         message_com::send_cmdu(slave_socket, cmdu_tx);
         break;
     }
+    case beerocks_message::ACTION_MONITOR_CHANNEL_DUMP_SCAN_RESULTS_REQUEST: {
+        auto request =
+            beerocks_header
+                ->addClass<beerocks_message::cACTION_MONITOR_CHANNEL_DUMP_SCAN_RESULTS_REQUEST>();
+        if (!request) {
+            LOG(ERROR) << "addClass cACTION_MONITOR_CHANNEL_DUMP_SCAN_RESULTS_REQUEST failed";
+            return false;
+        }
+
+        auto response_out = message_com::create_vs_message<
+            beerocks_message::cACTION_MONITOR_CHANNEL_DUMP_SCAN_RESULTS_RESPONSE>(
+            cmdu_tx, beerocks_header->id());
+        if (!response_out) {
+            LOG(ERROR)
+                << "Failed building cACTION_MONITOR_CHANNEL_DUMP_SCAN_RESULTS_RESPONSE message!";
+            return false;
+        }
+
+        bool result = mon_wlan_hal->channel_scan_dump_results();
+        LOG_IF(!result, ERROR) << "channel_scan_dump_results Failed";
+
+        response_out->success() = (result) ? 1 : 0;
+        message_com::send_cmdu(slave_socket, cmdu_tx);
+
+        auto notification = message_com::create_vs_message<
+           beerocks_message::cACTION_MONITOR_CHANNEL_SCAN_RESULTS_NOTIFICATION>(cmdu_tx);
+        if (!notification) {
+            LOG(ERROR) << "Failed building cACTION_MONITOR_CHANNEL_SCAN_RESULTS_NOTIFICATION msg";
+            return false;
+        }
+
+        //Sending the cACTION_MONITOR_CHANNEL_SCAN_RESULTS_NOTIFICATION without
```

Comment:
This is how we "trick" the task into moving to the next current state
Incoming: ACTION_MONITOR_CHANNEL_DUMP_SCAN_RESULTS_REQUEST
Outgoing:
- ACTION_MONITOR_CHANNEL_DUMP_SCAN_RESULTS_RESPONSE
- ACTION_MONITOR_CHANNEL_SCAN_RESULTS_NOTIFICATION

Most requests only return a single response, but this message sends one response and one notification immediately

Reply:
Why do we need a RESPONSE and a NOTIFICATION message?
Can't we use only a NOTIFICATION message?

When sending a response to a request, it'll contain the ID of the request.
When sending async notifications, the ID will remain 0.

Reply:
Since the response is part of the request-response flow and the notification has nothing to do it .
The notification is sent when the driver sends a `scan_triggered` NL event
But when using `dwell time = 0` the we never request a scan from the driver so the Nl event will never be sent

so to "trick" the FSM in the dcs' task to work as if a scan was triggered, we send the notification back manually while not actually triggering the bml channel scan